src/server/endpoints/postHandler.js

```diff
@@ -104,6 +104,31 @@ function dataValidation(req, res, postInfo) {
   return true;
 }
 
+function like(req, res) {
+  let id = req.params.id;
+  let token = req.headers.authorization;
+  let userName = null;
+  getAccessToken(token, function(err, item) {
+    if (err) {
+      res.status(HTTP_STATUSES.SERVER_ERROR)
+        .json({'errorType': 'server error'});
+      return;
+    }
+    if (item === null) {
+      res.status(HTTP_STATUSES.UNAUTHORIZED)
+        .json({'errorType': 'Unauthorized'});
+      return;
+    }
+    usersCollection.findUsername(item.userId, (result) => {
+      userName = result.username;
+      postsCollection.likePost(id, userName, function(data){
```

Comment:
Usually we using id-s when we are referencing to another entity. Please change username to userid